src/audio/codec_adapter/codec/cadence.c

```diff
@@ -99,7 +101,7 @@ static int cadence_codec_init(struct comp_dev *dev)
 	struct cadence_codec_data *cd = NULL;
 	uint32_t obj_size;
 	uint32_t no_of_api = ARRAY_SIZE(cadence_api_table);
-	uint32_t api_id = CODEC_GET_API_ID(codec->id);
+	uint32_t api_id = CODEC_GET_API_ID(DEFAULT_CODEC_ID);
```

Comment:
Why you do not set default CODEC_WRAPPER value at module initialization using module_id? It looks like with your change we lose cohesion between api_id and module_id?

Reply:
the idea is that module_id is not needed at all and all we need is the api_id. I've done that part in my other PR #5190 and I am working with @kkarask to fix the CI tests. I will modify this PR to change accordingly after that one is merged.